http/src/main/resources/jupyter/startup.sh

```diff
@@ -54,6 +57,22 @@ if [ "$UPDATE_WELDER" == "true" ] ; then
     docker-compose -f /etc/welder-docker-compose.yaml up -d
 fi
 
+# If a Jupyter start user script was specified, execute it now. It should already be in the docker container
+# via initialization in init-actions.sh (we explicitly do not want to recopy it from GCS on every cluster resume).
+if [ ! -z ${JUPYTER_START_USER_SCRIPT_URI} ] ; then
+  JUPYTER_START_USER_SCRIPT=`basename ${JUPYTER_START_USER_SCRIPT_URI}`
+  log 'Executing Jupyter user start script [$JUPYTER_START_USER_SCRIPT]...'
+  EXIT_CODE=0
+  docker exec --privileged -u root -e PIP_USER=false ${JUPYTER_SERVER_NAME} ${JUPYTER_HOME}/${JUPYTER_START_USER_SCRIPT} &> start_output.txt || EXIT_CODE=$?
+  if [ $EXIT_CODE -ne 0 ]; then
+    echo "User start script failed with exit code ${EXIT_CODE}. Output is saved to ${JUPYTER_START_USER_SCRIPT_OUTPUT_URI}"
+    retry 3 gsutil -h "x-goog-meta-passed":"false" cp start_output.txt ${JUPYTER_START_USER_SCRIPT_OUTPUT_URI}
+    exit $EXIT_CODE
+  else
+    retry 3 gsutil -h "x-goog-meta-passed":"true" cp start_output.txt ${JUPYTER_START_USER_SCRIPT_OUTPUT_URI}
```

Comment:
Similarly I don't think `retry` is in here

Reply:
Added